### DISCUSSION TITLE
4.0.0 update, unsure if bugs or intended changes but sharing anyway.

### DISCUSSION THREAD

**Discussion by OAK06** [tradingview/lightweight-charts#1265] [General] [answered]
Hi, I use the CDN version and I noticed today that my charts broke. 

From what I'm seeing, these are the parts that stopped working correctly:

When creating the chart I had this bit to change the theme:
```
layout: {
      backgroundColor: '#253248',
      textColor: 'rgba(255, 255, 255, 0.9)',
},
```
My chart displays the candles and then overlays a line chart at the bottom:
```
overlay: true,
scaleMargins: {
      top: 0.87,
      bottom: 0,
}
```
now for some reason it seems that the candles and the line chart switched places.

Working Chart (3.8.0):
------------------------
![image](https://user-images.githubusercontent.com/13987313/217580036-e817948b-a7dc-4d9f-9a6c-32c23640b263.png)

Broken Chart (4.0.0):
----------------------
![image](https://user-images.githubusercontent.com/13987313/217579016-41095a1d-2c55-43b1-b3ca-dec39abf36e2.png)

My apologies if these settings were changed intentionally.

**Comment by SlicedSilver** [comment, marked as answer]
These changes are intentional. We have a [migration guide for v3 to v4](https://tradingview.github.io/lightweight-charts/docs/migrations/from-v3-to-v4) and the release notes [here](https://github.com/tradingview/lightweight-charts/releases/tag/v4.0.0)

**Reply by OAK06** [reply]
Thank you! I'll work on migrating my code then.
